pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -164,15 +164,15 @@ type WorkflowSpec struct {
 	Templates []Template `json:"templates,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,1,opt,name=templates"`
 
 	// Entrypoint is a template reference to the starting point of the workflow.
-	Entrypoint string `json:"entrypoint,omitempty" protobuf:"bytes,2,opt,name=entrypoint"`
+	Entrypoint *string `json:"entrypoint,omitempty" protobuf:"bytes,2,opt,name=entrypoint"`
```

Comment:
I'm not clear why you would use `*string` here - as `string` has a zero value of "". Is that a valid entry point? I don't think so.